packages/datetime/test/dateRangePickerTests.tsx

```diff
@@ -904,47 +904,74 @@ describe("<DateRangePicker>", () => {
         });
 
         it("custom shortcuts set the displayed months correctly when start month changes", () => {
-            const dateRange = [new Date(2016, Months.JANUARY, 1), new Date(2016, Months.DECEMBER, 31)] as DateRange;
-            const { left, right } = render({
-                initialMonth: new Date(2015, Months.JANUARY, 1),
-                shortcuts: [{ label: "custom shortcut", dateRange }],
-            }).clickShortcut();
-            assert.isTrue(onChangeSpy.calledOnce);
-            left.assertMonthYear(Months.JANUARY, 2016);
-            right.assertMonthYear(Months.FEBRUARY, 2016);
-        });
+            const dateRange = [
+                new Date(2016, Months.JANUARY, 1, 10, 20, 30),
+                new Date(2016, Months.DECEMBER, 31, 10, 20, 30),
+            ] as DateRange;
 
-        it(
-            "custom shortcuts set the displayed months correctly when start month changes " +
-                "and contiguousCalendarMonths is false",
-            () => {
-                const dateRange = [new Date(2016, Months.JANUARY, 1), new Date(2016, Months.DECEMBER, 31)] as DateRange;
+            const test = (shouldChangeTime: boolean) => {
                 const { left, right } = render({
-                    contiguousCalendarMonths: false,
                     initialMonth: new Date(2015, Months.JANUARY, 1),
-                    shortcuts: [{ label: "custom shortcut", dateRange }],
+                    shortcuts: [{ label: "custom shortcut", dateRange, shouldChangeTime }],
                 }).clickShortcut();
                 assert.isTrue(onChangeSpy.calledOnce);
                 left.assertMonthYear(Months.JANUARY, 2016);
-                right.assertMonthYear(Months.DECEMBER, 2016);
+                right.assertMonthYear(Months.FEBRUARY, 2016);
+            };
+
+            test(true);
+            test(false);
```

Comment:
❌ these should be separate `it()` tests. please refactor this code so the time-changing tests run separately.

honestly i'd prefer if you 1) reverted this entire file and then 2) simply added a single test with `shouldChangeTime: true` to validate that the prop does what we think.